### PR TITLE
Use server-side filtering when retrieving Cloud Foundry logs

### PR DESCRIPTION
**Pull request by jsoriano**
## What does this PR do?

Use server-side filtering when retrieving Cloud Foundry logs.

## Why is it important?

It reduces CPU usage of Filebeat significatively.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] ~~I have made corresponding change to the default configuration files~~
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Author's Checklist

- [ ]

## How to test this PR locally

* Run Cloud Foundry input with `firehose_filter: logs`, check that log messages are still collected.

## Related issues

-

### CODE REVIEW

**Review by MichaelKatsoulis — APPROVED** [review]

File: x-pack/libbeat/common/cloudfoundry/dopplerconsumer.go [hunk 1]
```diff
@@ -80,23 +81,49 @@ func (c *DopplerConsumer) Run() {
...
+		return filterLogs, -1
+	case FirehoseFilterLogs:
+		// Uses filter-type=logs in requests to the firehose.
+		return filterLogs, consumer.LogMessages
+	case FirehoseFilterMetrics:
+		// Uses filter-type=metrics in requests to the firehose.
+		return filterNoFilter, consumer.Metrics
+	default:
+		// TODO: Handle unknown filters.
+		return filterNoFilter, -1
```
Comment (jsoriano) [inline x-pack/libbeat/common/cloudfoundry/dopplerconsumer.go:116]: We should decide if we want some kind of error handling here, or if we just consider `all` the default in this case.

File: x-pack/libbeat/common/cloudfoundry/dopplerconsumer.go [hunk 2]
```diff
@@ -80,23 +81,49 @@ func (c *DopplerConsumer) Run() {
...
+func selectFilter(firehoseFilter string) (func(*events.Envelope) bool, consumer.EnvelopeFilter) {
+	switch firehoseFilter {
+	case FirehoseFilterAll:
+		return filterNoFilter, -1
+	case FirehoseFilterAllLogs:
+		// Requests all events, and selects log-like events.
+		return filterLogs, -1
+	case FirehoseFilterLogs:
+		// Uses filter-type=logs in requests to the firehose.
+		return filterLogs, consumer.LogMessages
```
Comment (jsoriano) [inline x-pack/libbeat/common/cloudfoundry/dopplerconsumer.go:110]: LGTM to use both filters here.

File: x-pack/libbeat/common/cloudfoundry/dopplerconsumer.go [hunk 3]
```diff
@@ -80,23 +81,49 @@ func (c *DopplerConsumer) Run() {
 }
 
 func (c *DopplerConsumer) logsFirehose() {
-	c.firehose(c.callbacks.Log, consumer.LogMessages)
+	filter := FirehoseFilterAllLogs
```
Comment (jsoriano) [inline x-pack/libbeat/common/cloudfoundry/dopplerconsumer.go:84]: If we are sure that `FirehoseFilterAllLogs` and `FirehoseFilterLogs` collect the same logs, but `FirehoseFilterLogs` is much more efficient, wdyt about using `FirehoseFilterLogs` here and don't expose the option to the user?
```suggestion
	filter := FirehoseFilterLogs
```

**Review by jsoriano — COMMENTED** [review]
@MichaelKatsoulis @gizas if both logs modes collect the same logs, I wonder if we should just use the new method with server and client filtering, and don't expose the option to user. Wdyt?

**Comment by MichaelKatsoulis** [comment]
> @MichaelKatsoulis @gizas if both logs modes collect the same logs, I wonder if we should just use the new method with server and client filtering, and don't expose the option to user. Wdyt?

+1 on that. Let's not expose the `logs` vs `all-logs`. Let's keep `all-logs` with server and client side filtering , but expose the other options

**Comment by jsoriano** [comment]
> Let's not expose the `logs` vs `all-logs`. Let's keep `all-logs` with server and client side filtering , but expose the other options

And given that, shall we be exposing anything at all regarding this?

I have pushed a couple of changes, to remove `all-logs` vs. `logs` and to completely remove the setting and simplify the code, let me know what you think.

**Comment by MichaelKatsoulis** [comment]
@jsoriano it works as expected. We can merge when ci passes

**Pull request merged**